CondFormats/HGCalObjects/src/HGCalMappingModuleIndexerTrigger.cc

```diff
@@ -0,0 +1,195 @@
+  }
+  HGCalTriggerFEDReadoutSequence& frs = fedReadoutSequences_[fedid];
+  frs.id = fedid;
+
+  // asin position, resize if needed, and fill the type code
+  uint32_t idx = modFedIndexer_.denseIndex({{econtIdx, 0}});  // This will give back the econt*1+0 = econt
+  if (idx >= frs.readoutTypes_.size()) {
+    frs.readoutTypes_.resize(idx + 1, -1);
+  }
+  frs.readoutTypes_[idx] = typecodeIdx;
+
+  // std::count another typecodein the global list
+  if (typecodeIdx >= globalTypesCounter_.size()) {
+    globalTypesCounter_.resize(typecodeIdx + 1, 0);
+    globalTypesNTrLinks_.resize(typecodeIdx + 1, 0);
+    globalTypesNTCs_.resize(typecodeIdx + 1, 0);
+    offsetsTC_.resize(typecodeIdx + 1, 0);
+  }
+  globalTypesCounter_[typecodeIdx]++;
+  globalTypesNTrLinks_[typecodeIdx] = nTrLinks;
+  globalTypesNTCs_[typecodeIdx] = nTCs;
+
+  // add typecode string to map to retrieve fedId & modId later
+  if (!typecode.empty()) {
+    if (typecodeMap_.find(typecode) != typecodeMap_.end()) {     // found key
+      const auto& [fedid_, modid_] = typecodeMap_.at(typecode);  // (fedId,modId)
+      edm::LogWarning("HGCalMappingModuleIndexerTrigger")
+          << "Found typecode " << typecode << " already in map (fedid,modid)=(" << fedid_ << "," << modid_
+          << ")! Overwriting with (" << fedid << "," << idx << ")...";
+    }
+    LogDebug("HGCalMappingModuleIndexerTrigger")
+        << "HGCalMappingModuleIndexerTrigger::processNewModule: Adding typecode=\"" << typecode
+        << "\" with fedid=" << fedid << ", idx=" << idx << " (will be re-indexed after finalize)";
+    typecodeMap_[typecode] = std::make_pair(fedid, idx);
+  }
+}
+
+// @short to be called after all the modules have been processed
+void HGCalMappingModuleIndexerTrigger::finalize() {
+  // max indices at different levels
+  nfeds_ = fedReadoutSequences_.size();
+  maxModulesIdx_ = std::accumulate(globalTypesCounter_.begin(), globalTypesCounter_.end(), 0);
+  maxTrLinksIdx_ =
+      std::inner_product(globalTypesCounter_.begin(), globalTypesCounter_.end(), globalTypesNTrLinks_.begin(), 0);
+  maxNTCIdx_ = std::inner_product(globalTypesCounter_.begin(), globalTypesCounter_.end(), globalTypesNTCs_.begin(), 0);
+
+  // compute the global offset to assign per board type, eRx and channel data
+  offsetsModule_.resize(maxModulesIdx_, 0);
+  offsetsTrLink_.resize(maxModulesIdx_, 0);
+  offsetsTC_.resize(maxModulesIdx_, 0);
+  for (size_t i = 1; i < globalTypesCounter_.size(); i++) {
+    offsetsModule_[i] = globalTypesCounter_[i - 1];
+    offsetsTrLink_[i] = globalTypesCounter_[i - 1] * globalTypesNTrLinks_[i - 1];
+    offsetsTC_[i] = globalTypesCounter_[i - 1] * globalTypesNTCs_[i - 1];
+  }
+  std::partial_sum(offsetsModule_.begin(), offsetsModule_.end(), offsetsModule_.begin());
+  std::partial_sum(offsetsTrLink_.begin(), offsetsTrLink_.end(), offsetsTrLink_.begin());
+  std::partial_sum(offsetsTC_.begin(), offsetsTC_.end(), offsetsTC_.begin());
```

Comment:
Similar comment as above: since you are already doing the loop to compute the values to be summed, it would be simpler to just perform the sums inside the loop ?

Otherwise, if you prefer to use the STL algorithm approach, can you compute the sums from the existing collections without setting the indices by hand ?
(I really don't know, but I'm not fond of the STL approach, so I don't really know 🤷🏻)